portfolio/src/main/webapp/script.js

```diff
@@ -41,3 +43,21 @@ function createListElement(text) {
   liElement.innerText = text;
   return liElement;
 }
+
+/** Chooses what to display based on login status */
+function determineVisibleItems() {
+  fetch('/login').then(response => response.json()).then(loginInfo => {
+    const linkEl = document.getElementById('login-link');
+
+    // If the user is logged in
+    if(loginInfo.status) {
+      // Unhide comments
+      document.getElementById('hidden').style.display = 'block';
+      linkEl.innerHTML = 'You are currently logged in with your Google account. Log out <a href=\"' + loginInfo.url + '\">here</a>.';
+    }
+    // If the user is not logged in
```

Comment:
You are perhaps handling this case somewhere else (most probably through a redirect), but just in case you haven't considered, what happens to the comments when a user was logged in and was able to see the comments and then logged out?

Reply:
Not required. Thanks.